File: examples/javascript/dfrph.js
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */


var sensorObj = require('jsupm_dfrph');

// Instantiate a DFRPH sensor on analog pin A0, with an analog
// reference voltage of 5.0
var sensor = new sensorObj.DFRPH(0, 5.0);


// After calibration, set the offset (based on calibration with a pH
// 7.0 buffer solution).  See the UPM sensor documentation for
// calibrations instructions.
sensor.setOffset(0.065);

// Every second, sample the pH and output it's corresponding
// analog voltage.

setInterval(function()
{
    console.log("Detected volts: " + sensor.volts());
    console.log("pH value: " + sensor.pH());
}, 1000);

// exit on ^C
process.on('SIGINT', function()
{
    sensor = null;
    sensorObj.cleanUp();
    sensorObj = null;
    console.log("Exiting.");
    process.exit(0);
});

File: examples/c/dfrph.c
/*
 * Author: Noel Eck <noel.eck@intel.com>
 * Copyright (c) 2016 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <unistd.h>
#include <signal.h>

#include "dfrph.h"
#include "upm_utilities.h"

bool shouldRun = true;

void sig_handler(int signo)
{
    if (signo == SIGINT)
        shouldRun = false;
}

int main()
{
    signal(SIGINT, sig_handler);

    //! [Interesting]

    // Instantiate a dfrph sensor on analog pin A0
    dfrph_context sensor = dfrph_init(0);

    if (!sensor)
    {
        printf("dfrph_init() failed.\n");
        return(1);
    }

    // Every half a second, sample the sensor output
    while (shouldRun)
    {
        float volts = 0.0, pH = 0.0;

        dfrph_get_raw_volts(sensor, &volts);
        dfrph_get_ph(sensor, &pH);

        printf("Detected volts: %0.03f\n", volts);
        printf("pH value: %0.03f\n", pH);

        upm_delay_ms(500);
    }

    //! [Interesting]

    printf("Exiting\n");

    dfrph_close(sensor);

    return 0;
}
